frontend/test/specs/notification/pushNotificationServiceSpec.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+(fdescribe('Test PushNotificationService', function () {
+    var service, messaging, defaultToken, notificationsRef;
+
+    const fakeCallback = function fakeCallback(data) {
+        return {
+            then: function (callback) {
+                return callback(data);
+            }
+        };
+    };
+
+    beforeEach(module('app'));
+
+    beforeEach(inject(function ($firebaseArray, PushNotificationService, AuthService) {
+        service = PushNotificationService;
+        messaging = firebase.messaging();
+        var ref = firebase.database().ref();
+        AuthService.login(new User({key: 'opaksdapodkadpk'}));
+        notificationsRef = ref.child("notifications/key");
+        defaultToken = 'oaspkd-OPASKDAPO';
+    }));
+
+    describe('requestNotificationPermission', () => {
+        it('should call saveToken', () => {
+            spyOn(messaging, 'requestPermission').and.callFake(fakeCallback);
+            spyOn(messaging, 'getToken').and.callFake(fakeCallback);
+            spyOn(service, '_saveToken').and.callFake(fakeCallback);
+
+            service._requestNotificationPermission();
+
+            expect(messaging.requestPermission).toHaveBeenCalled();
+            expect(messaging.getToken).toHaveBeenCalled();  
+            expect(service._saveToken).toHaveBeenCalled();
+        });
+
+        it('should not call saveToken', () => {
```

Comment:
should not call, why?

Reply:
In 'it' i'm simulating the case when the user doesn't enable the notifications.

Reply:
I'm sorry I wasn't clear. Its good to describe in the 'it' what's the test about.
So it should be something like `should not call saveToken when the user doesn't enable the notifications`